paywall/src/data-iframe/Mailbox.ts

```diff
@@ -311,12 +313,15 @@ export default class Mailbox {
    * This is called by the BlockchainHandler when there is an update to chain data
    */
   emitChanges(newData: BlockchainData) {
+    const { keys, transactions } = newData
```

Comment:
Why not using the `getDataToSend` function below?

Reply:
As I understand it, there's a chance that we can have new keys before we have new locks. I think if we used `getDataToSend` we might hold off on sending the new keys until we have gotten the locks.

I think we need to look at the architecture here, since `getDataToSend` seems to be oriented entirely around the caching.